src/packaging/tags.py

```diff
@@ -37,7 +38,7 @@
 }
 
 
-_32_BIT_INTERPRETER = sys.maxsize <= 2**32
+_32_BIT_INTERPRETER = struct.calcsize("P") == 4
```

Comment:
If we also want to report 16-bit architecture as _32_BIT_INTERPRETER in order to behave more like that was when using sys.maxsize, then this could be replaced by
```suggestion
_32_BIT_INTERPRETER = struct.calcsize("P") < 8
```

Reply:
I don’t think CPython supports 16-bit; is there any known implementation that does?

Reply:
This might be something that does exist, I don't know.
I found [micropython ](https://micropython.org) that runs on micro-controllers but it seems support is for 32bit architectures minimum.

Reply:
I don't think this is worth worrying about until someone comes forward with a widely-used build of Python that's not 32 or 64-bit.